jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -917,227 +919,512 @@ ecma_number_to_int32 (ecma_number_t num) /**< ecma-number */
 } /* ecma_number_to_int32 */
 
 #if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT64
+
+#define GRISU3_SIGN_BIT                 0x8000000000000000ULL
+#define GRISU3_EXPONENT_MASK            0x7FF0000000000000ULL
+#define GRISU3_FRACTION_MASK            0x000FFFFFFFFFFFFFULL
+#define GRISU3_IMPLICIT_ONE             0x0010000000000000ULL
+#define GRISU3_EXPONENT_POSITION        52
+#define GRISU3_EXPONENT_BIAS            1075
+#define GRISU3_1_LOG2_10                0.30102999566398114 /* 1 / lg(10) */
+#define GRISU3_MIN_TARGET_EXP           -60
+#define GRISU3_32_BIT_MASK              0xFFFFFFFFULL
+#define GRISU3_ENCODED_FLOAT_FRACT_SIZE 64
+#define GRISU3_MIN_CACHED_EXP           -348
+#define GRISU3_CACHED_EXP_STEP          8
+
 /**
-  * Perform conversion of 128-bit binary representation of number
-  * to decimal representation with decimal exponent.
-  */
-static void
-ecma_number_helper_binary_to_decimal (ECMA_NUMBER_CONVERSION_128BIT_INTEGER_ARG (fraction_uint128), /**< mantissa */
-                                      int32_t binary_exponent, /**< binary exponent */
-                                      int32_t *out_decimal_exp_p) /**< [out] decimal exponent */
+ * Encoded floating point type
+ *
+ * Note:
+ *      Section 3. (ISBN: 978-1-4503-0019-3)
```

Comment:
This reference in imprecise. The ISBN just refers to the proceedings of a conference, while the section number is the part of a paper within that volume. Based on this information only, the relevant text is not/hard to be identified.

Proper reference to the paper is:
Florian Loitsch: Printing Floating-point Numbers Quickly and Accurately with Integers.
In Proceedings of the 31st ACM SIGPLAN Conference on Programming Language Design and Implementation (PLDI '10), pages 233-243.
Toronto, Ontario, Canada, June 5-10, 2010.
ACM (ISBN:978-1-4503-0019-3).

It might not be worth giving all the details at every function doc comment, but -- at least -- it should be listed at the beginning of the GRISU3 implementation. (And then, the ISBN number is not to be repeated all the time either.)